File: orig/Game/src/ScoreItem.java
import javax.microedition.lcdui.*;

public class ScoreItem extends GameObject {
    public int r = 10;
    public int x, y;
    public boolean is_enemy_bullet = false;
    public int score;

    public ScoreItem(int x, int y) {
	super();
        this.x = x;
        this.y = y;

	score += MyMath.rand(0, 30) * 10;
    }

    public void update() {
	y += 1;
	if (y > canvas.getHeight())
	    game.remove_object(this);
    }

    public void paint(Graphics g) {
        g.setColor(0x0000FF);
        g.drawString("ÅÙ", x, y, g.HCENTER | g.VCENTER);
    }
}

File: js/scoreItem.js
// ScoreItem.java port

import { GameObject } from './gameObject.js';
import { rand } from './math.js';

export class ScoreItem extends GameObject {
    constructor(game, x, y) {
        super(game);
        this.r = 10;
        this.x = x;
        this.y = y;
        this.score = rand(0, 30) * 10;
    }

    update() {
        this.y += 1;
        if (this.y > this.game.HEIGHT)
            this.game.removeObject(this);
    }

    paint(g) {
        g.setColor(0x0000FF);
        g.drawString('♪', this.x, this.y, 1 | 2); // HCENTER | VCENTER
    }
}
